### ISSUE TITLE
bug(android): sometimes the Android keyboard pops up at wrong size

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#10054]
![image](https://github.com/keymanapp/keyman/assets/4498365/d3cb5da0-b059-434d-973b-69801b17975f)

I have not yet been able to figure out exactly what causes this, but sometimes when the OSK pops up on Android, the keyboard shows at the wrong size. Rotating the device seems to be the only way to address this.

Seems like there is probably a race condition somewhere there.

Keyman 17.0.200-alpha-test-9973 (but also seen on 17.0.x-alpha)

**Comment by mcdurdin** [comment]
Do you like my search history there? 🤣

**Comment by mcdurdin** [comment]
This may be related to #9191?

**Comment by mcdurdin** [comment]
Step 1: adds some logs for when the keyboard is displayed, saved in-app, that can be downloaded via Android Studio or whatever. Then when I experience the problem, I'll go ahead and grab the logs.

**Comment by darcywong00** [comment]
I haven't been able to get a repro of this. But we can keep an eye and see if it's fixed by #10373. Visually, it does look like the OSK is in a landscape "height" even though the device is portrait.

**Comment by darcywong00** [comment]
> Changes in this pull request will be available for download in [Keyman version 17.0.245-alpha](https://keyman.com/downloads/releases/alpha/17.0.245)

@mcdurdin to retest with Alpha and see if he still gets a repro

**Comment by mcdurdin** [comment]
I am running latest alpha and so far have not experienced this. So will go ahead and close this as fixed.

**Comment by mcdurdin** [comment]
Sadly, this has not yet been resolved. I am now observing with Keyman 17.0.256-alpha.

Two sample screenshots provided. What I think is interesting in these two images is that in one of them, the whole reserved keyboard area is squashed, but in the other one, the reserved keyboard area is the correct size, but the keyboard is being rendered at the wrong size. Not sure if that is useful for trying to track the issue or not, but it is curious.

![image](https://github.com/keymanapp/keyman/assets/4498365/a54672ca-8397-40cc-a455-88cc0677f5af)
![image](https://github.com/keymanapp/keyman/assets/4498365/5d5fbdd5-c8cb-42f5-98ad-109c3e29581a)

**Comment by jahorton** [comment]
Are we leaving this issue open to track things in case of a re-occurrence, or can we go ahead and close it now given the most recent PR listed in the timeline before this?

**Comment by darcywong00** [comment]
Oh right - I think #11108 and #11134 added some stability to this. So will close and monitor if it happens again.

**Comment by mcdurdin** [comment]
Okay, I am on 17.0.301-beta at present, will upgrade soon and continue to monitor.

**Comment by mcdurdin** [comment]
Reopening as this is not yet fixed, see:
* #11484
* #11604

**Comment by jahorton** [comment]
From select comments on #11484:

> In 17.0.325, rotation seems fine, and it is only restore that seems to sometimes go wrong. So I think we need to leave the rotation handler alone and look at the restore handler.

All error cases reported during testing were in regard to the system keyboard, rather than anything in-app.

**Comment by jahorton** [comment]
While investigating, I thought it interesting how we have the main app and the system keyboard both call through `KMManager`'s `onConfigurationChanged` to `KMKeyboard`'s... despite it likely being called by the system itself.  `onConfigurationChanged` is an `@Override` of the `KMKeyboard`'s base class `WebView`'s version of the method, after all.

That led me on some spelunking through repo history, through which I noticed the following:
- the `KMKeyboard.onConfigurationChanged` method was defined before our Android code was in-repo:  see #126.
- We started enforcing alternate keyboard heights with #5201.
- _Very shortly_ thereafter, we had the first keyboard-sizing issue, #5211, with a number of related follow-ups following in close succession as noted on #5201's history.

If there's something we're doing "wrong", as opposed to an "Official Way It Should Be Done (tm)", we probably diverged from the "Official" path then.

Looking into more detail... it does appear that we're supposed to trigger _layout_ stuff in `onConfigurationChanged`, for sure.  As for updating based on size, though... I'm not sure we're using the ideal spot for that.
